src/constraints/outage_constraints.jl

```diff
@@ -302,23 +312,20 @@ function add_MG_storage_dispatch_constraints(m,p)
         )
     end
     
-    # Dispatch to MG electrical storage is no greater than inverter capacity
-    # and can't charge the battery unless binMGStorageUsed = 1
-    @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
-        m[:dvStoragePower]["ElectricStorage"] >= sum(m[:dvMGProductionToStorage][t, s, tz, ts] for t in p.techs.elec)
-    )
-    
-    # Dispatch from MG storage is no greater than inverter capacity
-    # and can't discharge from storage unless binMGStorageUsed = 1
-    @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
-        m[:dvStoragePower]["ElectricStorage"] >= m[:dvMGDischargeFromStorage][s, tz, ts]
-    )
-    
-    # Dispatch to and from electrical storage is no greater than power capacity
-    @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
-        m[:dvStoragePower]["ElectricStorage"] >= m[:dvMGDischargeFromStorage][s, tz, ts]
-            + sum(m[:dvMGProductionToStorage][t, s, tz, ts] for t in p.techs.elec)
-    )
+    # Dispatch to and from MG electrical storage is no greater than power capacity
+    if "ElectricStorage" in p.s.storage.types.dc_coupled
+        @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
+            m[:dvStoragePower]["ElectricStorage"] >= m[:dvMGDischargeFromStorage][s, tz, ts] / p.s.storage.attr["ElectricStorage"].inverter_efficiency_fraction
```

Comment:
I think I'm confused about why the if statement is needed here, whereas in many of the other constraints you were able to avoid if statements by adding flows indexed on ac_coupled to flows index on dc_coupled. And, why does the discharge from storage need to be divided by `inverter_efficiency_fraction` for the dc-coupled case? Shouldn't the discharge from storage be treated the same between ac- and dc-coupled cases, because in both cases the discharge passes through an inverter (and isn't that already captured in `dvMGDischargeFromStorage`)? 

I could be misunderstanding some of this, but just checking!